packages/rolldown/src/utils/normalize-hook.ts

```diff
@@ -1,25 +1,19 @@
 import type { ObjectHook } from '../plugin'
-import type { AnyFn, AnyObj } from '../types/utils'
 
-type NotFn<T> = T extends AnyFn ? never : T
-
-export function normalizeHook<H extends ObjectHook<AnyFn, AnyObj>>(
-  hook: H,
-): H extends ObjectHook<infer Handler, infer Options>
-  ? [Handler, NotFn<Options>]
+export function normalizeHook<T extends ObjectHook<Function>>(
+  hook: T,
+): T extends ObjectHook<infer A, infer B>
+  ? T extends Function
+    ? [A, {}]
```

Comment:
I refactor the function to make support `ObjectHook<string, {}>` for AddonHooks, and remove using `infer` make typing clear, the `@ts-expect-error` is not used.

Reply:
I see the pr, but it lost the ability to infer the type

Reply:
And it may cause more type errors in the down stream 
![image](https://github.com/user-attachments/assets/c56b7240-50c6-4863-9489-33d0007644ff)